### PR TITLE
Decode filenames from any URLs before using them

### PR DESCRIPTION
**Pull request by ClementPasteau**

### CODE REVIEW

File: newIDE/app/src/Utils/GDevelopServices/Project.spec.js [hunk 1]
```diff
@@ -2,88 +2,120 @@
...
-        extractFilenameFromProjectResourceUrl(
+        extractDecodedFilenameFromProjectResourceUrl(
           'https://project-resources.gdevelop.io/a9fe5bce-de39-4147-a669-93fc5cd69632/resources/some/other-folder/1/2/_/6ef87bc678921eb4bfa2d04e5dc6a16b75f7b239f3163d0c5efe64d4cc501711-Pea-Happy.png?some=parameter'
         )
       ).toBe('Pea-Happy.png');
     });
+    it('extracts filename from a gdevelop.io bucket URL, if encoded', () => {
+      expect(
+        extractDecodedFilenameFromProjectResourceUrl(
+          'https://project-resources.gdevelop.io/a9fe5bce-de39-4147-a669-93fc5cd69632/resources/some/other%20folder/1/2/_/6ef87bc678921eb4bfa2d04e5dc6a16b75f7b239f3163d0c5efe64d4cc501711-Pea Happy'
```
Comment (AlexandreSi) [inline newIDE/app/src/Utils/GDevelopServices/Project.spec.js:59]: Should it be `'https://project-resources.gdevelop.io/a9fe5bce-de39-4147-a669-93fc5cd69632/resources/some/other%20folder/1/2/_/6ef87bc678921eb4bfa2d04e5dc6a16b75f7b239f3163d0c5efe64d4cc501711-Pea%20Happy'`?

File: newIDE/app/src/Utils/GDevelopServices/Project.spec.js [hunk 2]
```diff
@@ -2,88 +2,120 @@
...
+      expect(extractDecodedFilenameFromProjectResourceUrl('test.png')).toBe(
         'test.png'
       );
-      expect(extractFilenameFromProjectResourceUrl('test.png?what')).toBe(
-        'test.png?what'
+      expect(
+        extractDecodedFilenameFromProjectResourceUrl('test.png?what')
+      ).toBe('test.png?what');
+      expect(extractDecodedFilenameFromProjectResourceUrl('test%20.png')).toBe(
+        'test%20.png'
```
Comment (AlexandreSi) [inline newIDE/app/src/Utils/GDevelopServices/Project.spec.js:118]: Why is it not `'test .png'`?

File: newIDE/app/src/Utils/GDevelopServices/Project.spec.js (hunk 1 above)
Comment (ClementPasteau) [reply newIDE/app/src/Utils/GDevelopServices/Project.spec.js:59]: yes indeed

File: newIDE/app/src/Utils/GDevelopServices/Project.spec.js (hunk 2 above)
Comment (ClementPasteau) [reply newIDE/app/src/Utils/GDevelopServices/Project.spec.js:118]: This one is a bit confusing. This function is used to extract almost anything, and if it doesn't match an known url, it just returns the input as is.
Do you know if this was done on purpose, or just a case that should throw an error?

File: newIDE/app/src/Utils/GDevelopServices/Asset.spec.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      expect(
+        extractDecodedFilenameWithExtensionFromPublicAssetResourceUrl(
+          'https://asset-resources.gdevelop.io/public-resources/Pack name/b540a2c4b3a4d9a856819eb522473380fa98291e2f59fcf8905d99649a5b179b_file-to-download'
+        )
+      ).toStrictEqual('file-to-download');
+    });
+    it('works if url is encoded', () => {
+      expect(
+        extractDecodedFilenameWithExtensionFromPublicAssetResourceUrl(
+          'https://asset-resources.gdevelop.io/public-resources/Pack%20name/b540a2c4b3a4d9a856819eb522473380fa98291e2f59fcf8905d99649a5b179b_file%20to%20download.png'
```
Comment (4ian) [inline newIDE/app/src/Utils/GDevelopServices/Asset.spec.js:30]: Make a test with spaces too? In the URL.

File: newIDE/app/src/Utils/GDevelopServices/Shop.js [hunk 4]
```diff
@@ -206,10 +206,11 @@ export const isProductAuthorizedResourceUrl = (url: string): boolean =>
   isPrivateAssetResourceAuthorizedUrl(url) ||
   isPrivateGameTemplateResourceAuthorizedUrl(url);
 
-export const extractFilenameWithExtensionFromProductAuthorizedUrl = (
-  url: string
+export const extractDecodedFilenameWithExtensionFromProductAuthorizedUrl = (
+  productAuthorizedUrl: string
 ): string => {
-  const urlWithoutQueryParams = url.split('?')[0];
+  const decodedUrl = decodeURIComponent(productAuthorizedUrl);
```
Comment (4ian) [inline newIDE/app/src/Utils/GDevelopServices/Shop.js:212]: Seems risky to use this function on a full URL. Consider doing it on the basename only

Comment (ClementPasteau) [reply newIDE/app/src/Utils/GDevelopServices/Shop.js:212]: oh yes, mb

**Review by 4ian — APPROVED** [review]

**Pull request merged**